src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -0,0 +1,218 @@
+    const resultCell = $$('div', { className: 'coveo-result-cell' }, this.buildCaption());
+    this.buildMissingTerms().forEach(term => {
+      if (term) {
+        resultCell.append(term.el);
+      }
+    });
+    if (resultCell.el.childElementCount > 1) {
+      return $$('div', { className: 'coveo-result-row' }, resultCell);
+    }
+  }
+
+  private buildCaption(): Dom {
+    return $$('span', { className: 'coveo-field-caption' }, this.options.caption);
+  }
+
+  private buildMissingTerms(): Dom[] {
+    const terms: Dom[] = this.missingTerms.map(term => {
+      if (this.hideMissingTermifFeaturedResults(term) || this.hideMissingTermIfWildcard(term)) {
+        return;
+      }
+      return this.makeTermClickableIfEnabled(term);
+    });
+    return terms;
+  }
+
+  private executeNewQuery(missingTerm: string = this.queryStateModel.get('q')) {
+    this.usageAnalytics.logSearchEvent<IAnalyticsIncludeMissingTerm>(analyticsActionCauseList.missingTermClick, {
+      missingTerm: missingTerm
+    });
+    this.queryController.executeQuery();
+  }
+
+  private makeTermClickableIfEnabled(term: string): Dom {
+    if (this.options.clickable) {
+      const termElement = $$('button', { className: 'coveo-missing-term clickable' }, term);
+      termElement.on('click', () => {
+        this.addTermForcedToAppear(term);
+        this.executeNewQuery(term);
+      });
+      return termElement;
+    } else {
+      return $$('span', { className: 'coveo-missing-term' }, term);
+    }
+  }
+
+  private createWordBoundryDelimitedRegex(term: string): RegExp {
+    return XRegExp(`${this.wordBoundary}(${term})${this.wordBoundary}`, 'g');
+  }
+
+  private hideMissingTermifFeaturedResults(term: string): boolean {
+    this.updateTermForcedToAppear();
+    return this.termForcedToAppear.indexOf(term) !== -1;
+  }
+
+  private hideMissingTermIfWildcard(term): boolean {
+    const query = this.queryStateModel.get('q');
+    const regxStarWildcard = XRegExp(`(\\*${term})|${term}\\*`);
+    const regxxQuestionMarkWildcard = XRegExp(`(\\?${term})|${term}\\?`);
+
+    const foundStar = this.queryController.getLastQuery().wildcards && regxStarWildcard.test(query);
+    const foundQuestionMark = this.queryController.getLastQuery().questionMark && regxxQuestionMarkWildcard.test(query);
+
+    return foundStar || foundQuestionMark;
+  }
+
+  private hideMissingTermIfNumberOfResult(element: HTMLElement) {
+    const allMissingTerms = $$(element).findAll('.coveo-missing-term');
+    if (allMissingTerms.length <= this.options.numberOfResults) {
+      return;
+    }
+    for (let index = this.options.numberOfResults; index < allMissingTerms.length; index++) {
+      allMissingTerms[index].style.display = 'none';
+    }
+    const showMore = $$(
+      'button',
+      { className: 'coveo-missing-term-show-more' },
+      `${allMissingTerms.length - this.options.numberOfResults} more...`
+    );
+    const resultCell = $$(element).find('.coveo-result-cell');
```

Comment:
Similar comment regarding the result-cell, it should not be creating it internally, and should not rely on it's existence.

If you need a container/wrapper, you can create one but don't use the `result-cell/result-row` classes here.